### PR TITLE
[AOT] Rerun FVP test incase of first attempt failure

### PR DESCRIPTION
**Pull request by lhutton1**
In light of the flaky failures seen when running these sorts of tests (issues: #10300), adding a second attempt if the first attempt fails to try to help reduce the number of failures observed in CI.

### CODE REVIEW

File: tests/python/contrib/test_ethosu/infra.py [hunk 1]
```diff
@@ -251,13 +251,23 @@ def verify_source(
...
+            interface_api,
+            workspace_byte_alignment=16,
+            data_linkage=AOTDataLinkage(section="ethosu_scratch", alignment=16),
+        )
+
+    # TODO(lhutton1) This is a quick and dirty work around to help temporarily reduce
+    # the flakyness of the tests. Will remove once #10300 is resolved.
+    try:
+        run_mod()
+    except AssertionError:
```
Comment (manupa-arm) [inline tests/python/contrib/test_ethosu/infra.py:268]: @lhutton1 thanks for quickly taking this up!

We need to print out the logs if it fails (even once) if we are ever to fix this. i.e. we let it rerun but we do print the logs.
Generally, printing out run_log and compile_log would give the sufficient info to debug this.

**Review by manupa-arm — CHANGES_REQUESTED** [review]
@lhutton1 lets print the logs even in a re-attempt

File: tests/python/contrib/test_ethosu/infra.py (hunk 1 above)
Comment (manupa-arm) [reply tests/python/contrib/test_ethosu/infra.py:268]: Or will it already happen by the time we land the other logging PR?

Comment (lhutton1) [reply tests/python/contrib/test_ethosu/infra.py:268]: I think we should try to land #10214 first, then rebase this patch and print the RuntimeError before the retry

**Review by manupa-arm — APPROVED** [review]
LGTM as a stop gap!

appreciate a look from original authors : @grant-arm @Mousius

**Comment by grant-arm** [comment]
Thanks for looking at this @lhutton1
LGTM!

**Pull request merged**